pkg/analytics/client.go

```diff
@@ -0,0 +1,232 @@
+// (C) Copyright 2020 Hewlett Packard Enterprise Development LP
+
+package analytics
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/HewlettPackard/hpecli/internal/platform/db"
+	"github.com/HewlettPackard/hpecli/internal/platform/rest"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+// TrackingID - Google Analytics tracking ID
+const TrackingID = "UA-159515478-1"
+
+// GAClientIDKey - Google Analytics Client ID Key
+const GAClientIDKey = "GA_CLIENT_ID"
+
+// DisableAnalyticsKey - maintain Google Analytics enable/disable status
+const DisableAnalyticsKey = "GA_DISABLE"
+
+const falseValue = "false"
+
+// Client - wrapper class for Google Analytics Measurement Protocol api's
+type Client struct {
+	Version            string
+	EventHitType       string
+	Eventcategory      string
+	EventAction        string
+	EventValue         string
+	EventLabel         string
+	UserAgent          string
+	ApplicationName    string
+	ApplicationVersion string
+	*rest.Request
+}
+
+// NewAnalyticsClient create
+func NewAnalyticsClient(version, eventHitType, eventCategory, eventAction,
+	eventValue, eventLabel, userAgent, applicationVersion,
+	applicationName string) *Client {
+	return &Client{
+		Version:            version,
+		EventHitType:       eventHitType,
+		Eventcategory:      eventCategory,
+		EventAction:        eventAction,
+		EventLabel:         eventLabel,
+		EventValue:         eventValue,
+		UserAgent:          userAgent,
+		ApplicationVersion: applicationVersion,
+		ApplicationName:    applicationName,
+	}
+}
+
+func clientID() (string, error) {
+	d, err := db.Open()
+	if err != nil {
+		logrus.Debug("Unable to open DB to get client ID")
+		return "", err
+	}
+
+	defer d.Close()
+
+	var ID string
+	if err := d.Get(GAClientIDKey, &ID); err != nil {
+		if errors.Is(err, db.ErrNotFound) {
+			// couldn't find existing Id, so generate one and store it
+			ID = newClientID()
+			logrus.Debugf("Didn't find existing clientID, generating a new one: %s", ID)
+			_ = d.Put(GAClientIDKey, ID)
+
+			return ID, nil
+		}
+		// unknown error getting key
+		return newClientID(), nil
+	}
+
+	return ID, nil
+}
+
+// Generates a new unique id
+func newClientID() string {
+	return uuid.New().String()
+}
+
+// enableGoogleAnalytics enable google analytics for HPE CLI
+func enableGoogleAnalytics() bool {
```

Comment:
Does return value indicate if it was able to enable it?
Seems like if it is enabled, and enable is called again .. and it fails it would report that it was disabled (which is wrong)
return error to indicate failure to change the value?